pyramid/tierFS.go

```diff
@@ -84,93 +88,71 @@ func NewFS(c *Config) (FS, error) {
 // 1. Adds stored files to the eviction control
 // 2. Remove workspace directories and all its content if it
 //	  exist under the namespace dir.
-func handleExistingFiles(eviction *lruSizeEviction, fsLocalBaseDir string) error {
-	var workspaceDirs []string
+func handleExistingFiles(eviction eviction, fsLocalBaseDir string) error {
 	if err := filepath.Walk(fsLocalBaseDir, func(rPath string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
 		if info.IsDir() {
 			if info.Name() == workspaceDir {
 				// skipping workspaces and saving them for later delete
-				workspaceDirs = append(workspaceDirs, rPath)
+				if err := os.RemoveAll(rPath); err != nil {
+					return fmt.Errorf("removing dir: %w", err)
+				}
 				return filepath.SkipDir
 			}
 			return nil
 		}
 
-		if !eviction.store(relativePath(rPath), info.Size()) {
-			// Cache rejected the file, can be removed from local storage.
-			// Should be relatively rare, as cache sum cost on startup should accurately
-			// represent previous cost limitations.
-			removeFromLocal(fsLocalBaseDir, relativePath(rPath))
-		}
+		eviction.store(relativePath(rPath), info.Size())
 		return nil
 	}); err != nil {
 		return fmt.Errorf("walking the fs dir: %w", err)
 	}
 
-	for _, dir := range workspaceDirs {
-		if err := os.RemoveAll(dir); err != nil {
-			return fmt.Errorf("removing dir: %w", err)
-		}
-	}
-
 	return nil
 }
 
 func (tfs *TierFS) removeFromLocal(rPath relativePath) {
-	removeFromLocal(tfs.fsLocalBaseDir, rPath)
+	// This will be called by the cache eviction mechanism during entry insert.
+	// We don't want to wait while the file is being removed from the local disk.
```

Comment:
Neat!  But it _does_ mean we will need to be very careful how we handle errors there.

Reply:
Nothing much we can do, right?
For now logging it, will add metrics later on.